portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,20 @@ function addRandomGreeting() {
   return greetingContainer.innerText = greeting;
 }
 
+/** Fetches the current date from the server and adds it to the page. */
+async function showServerTime() {
+  const responseFromServer = await fetch('/date');
+  const textFromResponse = await responseFromServer.text();
+
+  const dateContainer = document.getElementById('date-container');
+  dateContainer.innerText = textFromResponse;
+}
+
+async function addRandomString() {
+  const responseFromServer = await fetch('/string');
+  const textFromResponse = await responseFromServer.text();
+
+  const dateContainer = document.getElementById('string-container');
```

Comment:
It would be good to update this `dateContainer` variable name to reflect what it's being used for here. Maybe either just `container` or `stringContainer`.